app/clients/cloudwatch/aws_cloudwatch.py

```diff
@@ -0,0 +1,86 @@
+import json
+import re
+import time
+
+from boto3 import client
+
+from app.clients import Client
+from app.cloudfoundry_config import cloud_config
+
+
+class AwsCloudwatchClient(Client):
+    """
+    This client is responsible for retrieving sms delivery receipts from cloudwatch.
+    """
+
+    def init_app(self, current_app, *args, **kwargs):
+        self._client = client(
+            "logs",
+            region_name=cloud_config.sns_region,
+            aws_access_key_id=cloud_config.sns_access_key,
+            aws_secret_access_key=cloud_config.sns_secret_key
+        )
+        super(Client, self).__init__(*args, **kwargs)
+        self.current_app = current_app
+        self._valid_sender_regex = re.compile(r"^\+?\d{5,14}$")
+
+    @property
+    def name(self):
+        return 'cloudwatch'
+
+    def _get_all_logs(self, my_filter, log_group_name):
+
+        # Check all events in the last 30 minutes
+        now = round(time.time() * 1000)
+        beginning = now - 30 * 60 * 1000
```

Comment:
Could we set the beginning time to be the time that the message was created? That could prevent this running longer that it would need to.

Reply:
I'll set it to the time the notification was sent